test/e2e_node/runner/remote/run_remote.go

```diff
@@ -623,24 +623,48 @@ func createInstance(imageConfig *internalGCEImage) (string, error) {
 	}
 	i.Scheduling = &scheduling
 	i.Metadata = imageConfig.metadata
+	var insertionOperationName string
 	if _, err := computeService.Instances.Get(*project, *zone, i.Name).Do(); err != nil {
 		op, err := computeService.Instances.Insert(*project, *zone, i).Do()
+
 		if err != nil {
 			ret := fmt.Sprintf("could not create instance %s: API error: %v", name, err)
 			if op != nil {
 				ret = fmt.Sprintf("%s: %v", ret, op.Error)
 			}
 			return "", fmt.Errorf(ret)
 		} else if op.Error != nil {
-			return "", fmt.Errorf("could not create instance %s: %+v", name, op.Error)
+			var errs []string
+			for _, insertErr := range op.Error.Errors {
+				errs = append(errs, fmt.Sprintf("%+v", insertErr))
+			}
+			return "", fmt.Errorf("could not create instance %s: %+v", name, errs)
+
 		}
+		insertionOperationName = op.Name
 	}
-
 	instanceRunning := false
 	for i := 0; i < 30 && !instanceRunning; i++ {
 		if i > 0 {
 			time.Sleep(time.Second * 20)
 		}
+		var insertionOperation *compute.Operation
+		insertionOperation, err = computeService.ZoneOperations.Get(*project, *zone, insertionOperationName).Do()
+		if err != nil {
+			continue
+		}
+		if strings.ToUpper(insertionOperation.Status) != "DONE" {
+			err = fmt.Errorf("instance insert operation %s not in state DONE, was %s", name, insertionOperation.Status)
+			continue
+		}
+		if insertionOperation.Error != nil {
+			var errs []string
+			for _, insertErr := range insertionOperation.Error.Errors {
+				errs = append(errs, fmt.Sprintf("%+v", insertErr))
+			}
+			return name, fmt.Errorf("could not create instance %s: %+v", name, errs)
```

Comment:
To confirm, we are short circuiting the retry loop when we know the `insertionOperation` failed, because there's no way for that `insertionOperation` to come from failure to success, regardless of how long we wait, correct?

Reply:
Also, to be sure I'm understanding - the main goal of this PR is that now we will actually know the reasons why the `insertionOperation` failed, instead of simply knowing that the instance isn't running?

Reply:
> To confirm, we are short circuiting the retry loop when we know the insertionOperation failed, because there's no way for that insertionOperation to come from failure to success, regardless of how long we wait, correct?

yes, afaik. Guess some of the gcloud folks have some more background info if that is interesting.

> Also, to be sure I'm understanding - the main goal of this PR is that now we will actually know the reasons why the insertionOperation failed, instead of simply knowing that the instance isn't running?

Yes! The old `404 instance not found` doesn't tell anything about the reason, making it hard for developers to debug, causing huge delays in fixing the _real_ problem. :smile: